test/rekt/features/trigger/feature.go

```diff
@@ -67,10 +69,31 @@ func TriggerDependencyAnnotation() *feature.Feature {
 		trigger.WithAnnotations(annotations),
 		trigger.WithBrokerName(brokerName),
 	}
-
 	// Install the trigger
 	f.Setup("install trigger", trigger.Install(triggerName, cfg...))
 
+	// trigger is not ready since the pingsource dependency is not installed yet
+	f.Setup("trigger is not ready before pingsource dependency exists", func(ctx context.Context, t feature.T) {
+		trigger.IsNotReady(triggerName)(ctx, t)
+
+		// verify that trigger's pingsource DependencyDoesNotExist condition is met
+		err := k8s.WaitForResourceCondition(ctx, t, environment.FromContext(ctx).Namespace(), triggerName, trigger.GVR(),
+			func(obj duckv1.KResource) bool {
+				expectedReason := "DependencyDoesNotExist"
+				expectedMessage := fmt.Sprintf(`Dependency does not exist: pingsources.sources.knative.dev "%s" not found`, psourcename)
+
+				condition := obj.Status.GetCondition(eventingv1.TriggerConditionDependency)
+				if condition != nil && condition.Reason == expectedReason && condition.Message == expectedMessage {
+					return true
+				}
+				return false
+			})
+
+		if err != nil {
+			t.Error("trigger's pingsource DependencyDoesNotExist condition not met", err)
+		}
```

Comment:
I would refactor this into it's own `StepFn`. Then the setup phase could be simpler and just have:

```go
f.Setup("trigger is not ready before pingsource dependency exists", trigger.IsNotReady)
f.Setup("trigger has DependencyDoesNotExist condition", trigger.DependencyDoesNotExist)
```